cpp/include/cudf/utilities/type_dispatcher.hpp

```diff
@@ -78,6 +86,11 @@ using id_to_type = typename id_to_type_impl<Id>::type;
     return Id;                                  \
   }                                             \
   template <>                                   \
+  inline std::string                            \
+  type_to_name::operator()<Type>() {            \
```

Comment:
As a function:
```
template <>
constexpr inline std::string type_to_name<Type>(){
   return CUDF_STRINGIFY(Type);
}
```

Reply:
It's used with a type dispatcher like so: https://github.com/rapidsai/cudf/pull/3237/files#diff-5efe9fb479e74208dfcc83f741af53b0R40

The functor needs to exist. Better it exists in one place than two.